system/include/compat/avxintrin.h

```diff
@@ -11,14 +11,505 @@
 #error "AVX instruction set not enabled"
 #endif
 
+#include <emmintrin.h>
+#include <emscripten/console.h>
 #include <nmmintrin.h>
+#include <pmmintrin.h>
+#include <smmintrin.h>
+#include <tmmintrin.h>
+#include <xmmintrin.h>
 
-static __inline__ __m128 __attribute__((__always_inline__, __nodebug__))
-_mm_broadcast_ss(const float *__mem_addr)
-{
-  return (__m128)wasm_v32x4_load_splat(__mem_addr);
+typedef struct {
+  __m128d v0;
+  __m128d v1;
+} __m256d;
+
+typedef struct {
+  __m128 v0;
+  __m128 v1;
+} __m256;
+
+typedef struct {
+  __m128i v0;
+  __m128i v1;
+} __m256i;
+
+typedef long long __m128i_u;
+typedef struct {
+  __m128i_u v0;
+  __m128i_u v1;
+} __m256i_u;
+
+union m256_data {
+  __m256i int_view;
+  __m256d double_view;
+  __m256 float_view;
+  __m128i_u int_u_view;
+};
+
+#define UNIMPLEMENTED(name)                                                    \
+  emscripten_err("warning: unsupported avx intrinsic: " #name "\n")
```

Comment:
Should this be a compiler error instead? 

If not that perhaps `abort()`..?

Reply:
A compile time diagnostic would be nice (ideally one that could be tuned to be a warning or an error, although I think we currently don't have such facilities possible).

If compile time diagnostic is not possible, then at runtime it would be good to not be an abort, but only a warning. This is because if it was an aborting error, it would throw people into a repeated find-first-error, fix, rebuild, retest loop to find and fix all sources of unused avx intrinsic use.

Reply:
How can the program meaningfully continue after trying to execute a missing intrinsic?  Is there some way to gracefully fall back to something else?

Reply:
By continuing to produce garbage (as the missing functions are no-ops) and emitting warnings from subsequent missing fucntions as it executes.

That way a developer would be able to see warning messages of multiple functions missing at one go, rather than having to do that tedious/neverending loop of fixing first error at abort, then rebuilding and retesting, to find the next such source of unimplemented function being executed. It is the same iteration problem that the safe_heap mode suffers from.

That being said, the only places where that UNIMPLEMENTED macro is used is in functions `_mm256_zeroall` and `_mm256_zeroupper`, and those can safely be no-op functions anyway, so this whole UNIMPLEMENTED macro can then be removed altogether.